### PR TITLE
feat: check (ban) architectually unwanted dependencies

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: misc/git-hooks/pre-commit [hunk 1]
```diff
@@ -33,6 +33,13 @@ fi
 export git_ls_files
 git_ls_files="$(git ls-files)"
 
+function check_forbiden_depenencies() {
+  if grep -E "(fedimint-mint|fedimint-wallet|fedimint-ln)" fedimint-server/Cargo.toml >&2 ; then
+    >&2 echo "fedimint-server/Cargo.toml must not depend on modules"
+    return 1
+  fi
+}
+export -f check_forbiden_depenencies
```
Comment (elsirion) [inline misc/git-hooks/pre-commit:42]: Can we do the same for others like new client <-> server?

**Review by elsirion — APPROVED** [review]

File: misc/git-hooks/pre-commit [hunk 2]
```diff
@@ -33,6 +33,13 @@ fi
 export git_ls_files
 git_ls_files="$(git ls-files)"
 
+function check_forbiden_depenencies() {
```
Comment (okjodom) [inline misc/git-hooks/pre-commit:36]: nit: *dependencies

File: misc/git-hooks/pre-commit (hunk 1 above)
Comment (dpc) [reply misc/git-hooks/pre-commit:42]: Which crates shouldn't include which exactly?

Comment (okjodom) [reply misc/git-hooks/pre-commit:42]: Citing recent example, `gateway/*` should not include `fedimint-server`. #1723

**Comment by jkitman** [comment]
@dpc can we enforce something similar to this? https://github.com/fedimint/fedimint/blob/master/docs/modular-architecture.md

**Comment by dpc** [comment]
> @dpc can we enforce something similar to this? https://github.com/fedimint/fedimint/blob/master/docs/modular-architecture.md

We can ban unwanted deps, but that's not exactly the same as enfocing that chart.

I added all the suggestions. We can always add more as we spot them.

**Review by okjodom — APPROVED** [review]
🚀

**Review by justinmoon — APPROVED** [review]

**Pull request merged**